### PR TITLE
Fix reading list item count issue #14469

### PR DESCRIPTION
**Pull request by haoxilu**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- https://github.com/forem/forem/issues/14469 reports `The reading list item count (in the title) isn't reactive`
- The issue was able to reproduce. The reason is that there was forgot to update`itemsTotal` after archived or unarchived. I just make minor changes to get the bug fixed.

## Related Tickets & Documents
Closes https://github.com/forem/forem/issues/14469

## QA Instructions, Screenshots, Recordings

- Reproduce the bugs https://github.com/forem/forem/issues/14469
> The reading list item count won't change.
CleanShot 2021-08-11 at 19 40 23

- Fixed. The reading list item count is changed after archived and unarchived.
CleanShot 2021-08-11 at 19 47 42

### UI accessibility concerns?

No. There are no UI changes.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Review by citizen428 — APPROVED** [review]

File: app/javascript/readingList/readingList.jsx [hunk 1]
```diff
@@ -117,6 +117,7 @@ export class ReadingList extends Component {
     this.setState({
       archiving: true,
       items: newItems,
+      itemsTotal: this.state.itemsTotal - 1,
```
Comment (rhymes) [inline app/javascript/readingList/readingList.jsx:120]: Nitpick: could you extract `itemsTotal` from `state` on line 109?

**Review by rhymes — COMMENTED** [review]
Thank you very much for the fix @haoxilu, I left a note about a stylistic change, let me know if you can apply it.

It works great!

File: app/javascript/readingList/readingList.jsx (hunk 1 above)
Comment (haoxilu) [reply app/javascript/readingList/readingList.jsx:120]: cool. I'll apply the change.

**Review by rhymes — APPROVED** [review]

**Comment by cmgorton** [comment]
@haoxilu thanks again for working on this bug. We are wrapping up the bug smash event and I wanted to encourage you to write a post about your experience if you have time.
You can use this template.
https://dev.to/new/devbugsmash

**Pull request merged**